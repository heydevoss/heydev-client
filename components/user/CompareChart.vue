<template>
  <vue-no-ssr>
    <div>
      <div>
        <chartjs-bar
          class="chart"
          :beginzero="true"
          :datasets="datasets"
          :labels="labels"
          :option="options"
        />
      </div>
    </div>
  </vue-no-ssr>
</template>

<script>
export default {
  props: {
    prContributor: Number,
    issuesContributor: Number,
    commitsContributor: Number,
    prOrganization: Number,
    issuesOrganization: Number,
    commitsOrganization: Number
  },
  data() {
    return {
      beginZero: true,
      labels: ['Pull Requests', 'Issues', 'Commits'],
      datasets: [
        {
          backgroundColor: '#4ECDC4',
          data: [
            this.prContributor,
            this.issuesContributor,
            this.commitsContributor
          ],
          label: 'Contributor'
        },
        {
          backgroundColor: '#FF6B6B',
          data: [
            this.prOrganization,
            this.issuesOrganization,
            this.commitsOrganization
          ],
          label: 'Organization'
        }
      ],
      options: {
        maintainAspectRatio: true,
        legend: {
          position: 'right'
        }
      }
    }
  }
}
</script>

<style scoped>
.chart {
  width: 500px;
  height: 500px;
}

@media screen and (max-width: 820px) {
  .chart {
    width: 300px;
    height: 300px;
  }
}
</style>
