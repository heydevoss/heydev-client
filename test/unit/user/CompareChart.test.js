import { shallowMount } from '@vue/test-utils'
import CompareChart from '@/components/user/CompareChart.vue'

describe('<CompareChart/>', () => {
  const contributor = {
    pr: 20,
    issues: 13,
    commits: 5
  }
  const org = {
    pr: 20,
    issues: 13,
    commits: 5
  }

  const factory = props =>
    shallowMount(CompareChart, {
      propsData: {
        ...props
      },
      stubs: {
        'chartjs-bar': '<div class="chartjs-bar-test"></div>',
        'vue-no-ssr': true
      }
    })

  const wrapper = factory({
    prContributor: contributor.pr,
    issuesContributor: contributor.issues,
    commitsContributor: contributor.commits,
    prOrganization: org.pr,
    issuesOrganization: org.issues,
    commitsOrganization: org.commits
  })

  test('should be a Vue instance and render correctly', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render the chartjs-bar', () => {
    expect(wrapper.html().includes('<div class="chartjs-bar-test"></div>'))
  })

  it('should contain contributor data', () => {
    expect(wrapper.vm.prContributor).toBe(contributor.pr)
    expect(wrapper.vm.issuesContributor).toBe(contributor.issues)
    expect(wrapper.vm.commitsContributor).toBe(contributor.commits)
  })

  it('should contain organization data', () => {
    expect(wrapper.vm.prOrganization).toBe(org.pr)
    expect(wrapper.vm.issuesOrganization).toBe(org.issues)
    expect(wrapper.vm.commitsOrganization).toBe(org.commits)
  })
  it('should contain Pull Requests, Issues and Commits labels', () => {
    expect(wrapper.vm.labels.includes('Pull Requests'))
    expect(wrapper.vm.labels.includes('Issues'))
    expect(wrapper.vm.labels.includes('Commits'))
  })
  it('should render when no props are passed', () => {
    const emptyWrapper = factory()
    expect(emptyWrapper.isVueInstance()).toBeTruthy()
  })
})
