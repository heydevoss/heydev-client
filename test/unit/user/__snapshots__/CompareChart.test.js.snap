// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CompareChart/> should be a Vue instance and render correctly 1`] = `
<vue-no-ssr-stub>
  <div>
    <div>
      <div
        beginzero="true"
        class="chartjs-bar-test chart"
        datasets="[object Object],[object Object]"
        labels="Pull Requests,Issues,Commits"
        option="[object Object]"
      />
    </div>
  </div>
</vue-no-ssr-stub>
`;
